pyproject.toml

```diff
@@ -48,30 +48,37 @@ packages = [
 [tool.poetry.dependencies]
 python = "^3.10"
 maturin = "^1.4.0"
-ibis-framework = {version = "9.0.0", extras = ["duckdb"]}
+ibis-framework = "9.0.0"
 dask = "^2023.12.1"
-xgboost = ">=1.6.1"
 attrs = "^23.2.0"
 connectorx = "^0.3.2"
-scikit-learn = "^1.4.0"
 psycopg2-binary = "^2.9.9"
 sqlalchemy = "^2.0.25"
-pytest-mock = "^3.14.0"
 pyarrow = "13.0.0"
-
+duckdb = {version = "^0.10.3", optional = true}
+snowflake-connector-python = {version = "^3.10.1", optional = true}
+datafusion = {version = "^34.0.0", optional = true}
```

Comment:
Do we need datafusion other than in tests?

Reply:
a user might want to have datafusion available as an engine, similar to duckdb and snowflake